widget.py

```
@@ -65,16 +142,21 @@ def nest(cls, region_cls):
         return region_cls.locator()
 
 
-class BaseInteractiveWidget(BaseWidget):
+class BaseUIWidget(BaseLogicalWidget):
     @property
     def is_displayed(self):
-        return self.browser.is_displayed(self)
+        return self.root.is_displayed
 
+    @property
+    def text(self):
+        return self.root.text
 
-class Input(BaseInteractiveWidget):
-    def __init__(self, browser, id):
-        super(Input, self).__init__(browser)
+
+class Input(BaseUIWidget):
+    def __init__(self, parent, id):
+        super(Input, self).__init__(parent)
```

Why all when you extend functionality of the child class? I guess it is better to explode on an extra arg than silently ignore something ...